### DISCUSSION TITLE
Does Watchtower updates itself when using exclusively the --run-once mode ?

### DISCUSSION THREAD

**Discussion by Antiz96** [containrrr/watchtower#1302] [Q&A] [answered]
Hi,

I'm personally not a huge fan of automatic update processes for various reasons. 
However, I like to simplify them as much as possible ! This is where the --run-once mode of watchtower is becoming handy (seriously that's an awesome option, thanks a lot for that :+1:).

I read in #565 and in #1157 that it indeed updates itself when using the "regular" mode, which makes sense as it monitors itself.

But what if you use it exclusively with the --run-once mode like I do ?
More simply, is the following command enough to ensure I'm always running the latest version of watchtower...
```
sudo docker run --rm -v /var/run/docker.sock:/var/run/docker.sock containrrr/watchtower --cleanup --run-once
```
...or do I need to run the following one instead ?
```
sudo docker pull containrrr/watchtower && sudo docker run --rm -v /var/run/docker.sock:/var/run/docker.sock containrrr/watchtower --cleanup --run-once
```

Thanks in advance for your response and thank you even more for your awesome work ! :smiley:

**Comment by simonaronsson** [comment, marked as answer]
No, it won't. The second snippet however should be sufficient. Good luck! 👍🏼

**Reply by Antiz96** [reply]
Alright, thank you !
